docs: update postgresql guide for new features (#197)
GreptimeTeam/docs: docs/user-guide/supported-protocols/postgresql.md

# PostgreSQL

> PostgreSQL wire protocol is supported by GreptimeDB, which means you can use

Reviewer: ```suggestion
GreptimeDB supports PostgreSQL wire protocol, which allows you to execute queries after connecting DB instances with the standard PostgreSQL client.
```

> standard PostgreSQL client to connect to DB instances and execute

Reviewer: delete this sentence if above is approved

queries. Note that when writing SQL queries, GreptimeDB follows MySQL dialect
instead of PostgreSQL.

> Both simple and extended query subprotocols are supported, except for those

Reviewer: ```suggestion
GreptimeDB supports simple and extendable query subprotocols, except parameter inference 
```

Author: here extended query is a name for subprotocol. but let reorganise later part of the sentence.

> clients relies on parameter type inference, like rust-postgres, will be

Reviewer: ```suggestion
like rust-postgres, which will be 
```

> supported later.

Reviewer: ```suggestion
supported in future versions.
```

Author: by the way you can select multiple lines when adding suggestions

## Using psql

> Currently we have PostgreSQL simple query subprotocol supported on

Reviewer: ```suggestion
As for PostgreSQL, only the simple query subprotocol is supported by 
```

Author: I just removed this line because we don't have to emphases the simple query support any more, as we already supported both

## Using database connectors

### Java

> You can use Postgres JDBC connector by specifying host, port, dbname. Username

Reviewer: ```suggestion
You can use the Postgres JDBC connector by specifying host, port, dbname. 
```

> and password are also supported when you have authentication enabled.

Reviewer: ```suggestion
If authentication is enabled, then you can also specify username and password. 
```

```java
String url = "jdbc:postgresql://localhost:4003/public";
Connection conn = DriverManager.getConnection(url);
```

### Python

> GreptimeDB now works with both [psycopg](https://www.psycopg.org/docs/) and its

Reviewer: ```suggestion
GreptimeDB is compatible with both [psycopg](https://www.psycopg.org/docs/) and
```

> legacy version, psycopg2. When using psycopg, remember to turn on `autocommit`.

Reviewer: ```suggestion
 psycopg2. When using psycopg, don't forget to turn on `autocommit`.
```
